addons/account/models/account_move.py

```diff
@@ -2540,6 +2542,30 @@ def _get_quick_edit_suggestions(self):
             self.quick_edit_total_amount - self.amount_total)['total_excluded']
         return {'account_id': account_id, 'tax_ids': taxes.ids, 'price_unit': price_untaxed}
 
+    @api.onchange('quick_edit_mode', 'journal_id', 'company_id')
+    def _quick_edit_mode_suggest_invoice_date(self):
+        for record in self:
+            if record.invoice_date:
+                continue
+            if record.quick_edit_mode:
+                # Suggest the Customer Invoice/Vendor Bill date based on previous invoice and lock dates
+                invoice_date = date.today()
+                prev_move = self.search([('state', '=', 'posted'),
+                                         ('journal_id', '=', record.journal_id.id),
+                                         ('company_id', '=', record.company_id.id),
+                                         ('invoice_date', '!=', False)],
+                                        limit=1)
+                if prev_move:
+                    lock_date = prev_move.company_id._get_user_fiscal_lock_date()
+                    prev_date_last_day_of_month = prev_move.invoice_date + relativedelta(day=31)
+                    if lock_date and prev_date_last_day_of_month <= lock_date:
+                        invoice_date = lock_date + relativedelta(months=1) + relativedelta(day=31)
+                    elif prev_date_last_day_of_month < date.today():
+                        invoice_date = prev_date_last_day_of_month
```

Comment:
So `_get_accounting_date` like suggested before doesn't work here?
You only take one lock date into account still, and you don't care about the format of the sequence (monthly/yearly)

Reply:
Nope, the computation are different.
For instance this test case returns 17/06/2022 instead of 31/03/2022.
```py
# We are June 17th. No Lock date. Bill Date of the most recent Vendor Bill : March 15th
# ==> Default New Vendor Bill date = March 31st (last day of March)
```
Or perhaps should `_get_accounting_date` be adapted (don't think so) ?

Reply:
The computation is slightly different because the logic is different. Calculating the most appropriate Accounting depending on the Invoice/Bill date (`_get_accounting_date`) is not the same as proposing a default Invoice/Bill date.

Reply:
> ```python
> # We are June 17th. No Lock date. Bill Date of the most recent Vendor Bill : March 15th
> # ==> Default New Vendor Bill date = March 31st (last day of March)
> ```

That's what I would expect from the Algorithme-des-bills

Reply:
Actually if we use the previous move's invoice date (if it exists) instead of today's, it works :)

Reply:
Well yes, that was the idea :sweat_smile: